### PR TITLE
Add initial support for TypeVarLike default parameters (PEP 696)

### PR DESCRIPTION
**Pull request by cdce8p**
This adds the `default` parameter to `TypeVar`, `ParamSpec`, and `TypeVarTuple`.
PR: https://github.com/python/typing_extensions/pull/77

https://peps.python.org/pep-0696/

/CC: @Gobot1234, @JelleZijlstra

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Thanks! We'll need to wait for the runtime PR to be merged before we can merge this :)

File: stdlib/typing_extensions.pyi
Comment (JelleZijlstra) [inline stdlib/typing_extensions.pyi:270]:
```suggestion
        default: Any | None = ...,
        infer_variance: bool = ...,
```
Might as well add this one too.

File: stdlib/typing_extensions.pyi
Comment (JelleZijlstra) [inline stdlib/typing_extensions.pyi:254]:
```suggestion
# The `default` parameter was added to TypeVar, ParamSpec, and TypeVarTuple (PEP 696)
# The `infer_variance` parameter was added to TypeVar (PEP 695)
```

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Actually it looks like we need to add a few more things to get CI green with the new release. I'll just hijack this PR to also add override and Any.

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**
